src/lineChartD3.ts

```diff
@@ -256,79 +228,81 @@ export const lineChartD3 = ((): IChartAdaptor<ILineChartProps> => {
+      const valuesCount = data.reduce((a: number, b): number => {
+        return b.data.length > a ? b.data.length : a;
+      }, 0)
       const w = gridWidth(props);
-      let yDomain;
-      let xDomain;
+      const h = gridHeight(props);
       const ys: any[] = [];
       const xs: any[] = [];
-      const yAxis = axisLeft<number>(yScale).ticks(axis.y.ticks);
 
-      const yTickSize = get(axis, 'y.tickSize', undefined);
-      if (yTickSize !== undefined) {
-        yAxis.tickSize(yTickSize);
-      }
-      if (shouldFormatTick(axis.y)) {
-        yAxis.tickFormat(formatTick(axis.y) as any);
-      }
+      const yAxis = axisLeft<number>(yScale);
+      const xAxis = axisBottom<number | string>(xScale);
 
-      const xAxis = axisBottom<number | string>(xScale)
-        .ticks(axis.x.ticks);;
+      ticks({
+        axis: xAxis,
+        valuesCount,
+        axisLength: w,
+        axisConfig: axis.x,
+        scaleBand: xScale,
+        limitByValues: true,
+      });
 
-      if (shouldFormatTick(axis.x)) {
-        xAxis.tickFormat(formatTick(axis.x) as any);
-      }
+      ticks({
+        axis: yAxis,
+        valuesCount,
+        axisLength: h,
+        axisConfig: axis.y,
+        scaleBand: yScale,
+      });
 
-      const xAxisHeight = getXAxisHeight(axis);
       const yAxisWidth = getYAxisWidth(axis);
 
       data.forEach((datum) => {
         datum.data.forEach((d) => {
           let parsedY = d.y;
           let parsedX = d.x;
           if (axis.y.scale === 'LOG' && d.y === 0) {
-            parsedY = ZERO_SUBSITUTE;
+            parsedY = ZERO_SUBSTITUTE;
           }
           if (axis.x.scale === 'LOG' && d.x === 0) {
-            parsedX = ZERO_SUBSITUTE;
+            parsedX = ZERO_SUBSTITUTE;
           }
           ys.push((parsedY));
-          xs.push((parsedX));
+          xs.push((typeof parsedX === 'number' ? applyDomainAffordance(parsedX) : parsedX));
         });
       });
-      yDomain = extent(ys);
-      xDomain = extent(xs);
+      const yDomain = extent(ys);
+      yDomain[0] = applyDomainAffordance(yDomain[0], false);
+      yDomain[1] = applyDomainAffordance(yDomain[1]);
+
+      const xDomain = extent(xs);;
       // domain mustn't be 0 as log(0) gives Infinity. 1 lower domain gives better looking graphs
-      if (axis.y.scale === 'LOG' && yDomain[0] === ZERO_SUBSITUTE) {
+      if (axis.y.scale === 'LOG' && yDomain[0] === ZERO_SUBSTITUTE) {
         yDomain[0] = 1;
       }
-      if (axis.x.scale === 'LOG' && xDomain[0] === ZERO_SUBSITUTE) {
+      if (axis.x.scale === 'LOG' && xDomain[0] === ZERO_SUBSTITUTE) {
         xDomain[0] = 1;
       }
       xScale
         .domain(xDomain)
         .rangeRound([0, w]);
-
       yScale.domain(yDomain)
-        .range([Number(height) - xAxisHeight, 0]);
+        .range([h, 0]);
 
       yAxisContainer
-        .attr('transform', `translate(${yAxisWidth}, 0)`)
+        .attr('transform', `translate(${yAxisWidth}, ${0})`)
```

Comment:
you've wrapped the 0 here in a literal template, did you mean it to be another value?

Reply:
nope just had it as a var before hand but it  should just be `0` - so updating